client/src/services/categories.service.ts

```diff
@@ -1,7 +1,25 @@
 import axiosInstance from './axios.service';
+import { store } from "../store/store";
+import { Gender } from "../store/auth/auth.model";
+
+export interface Category {
+    imageUrl: string
+    items: []
+    name: string
+    sentenceBeginning: string
+    _id: string
+}
 
 export const getCategories = () => {
     return axiosInstance.get(`/categories`).then((response) => {
-        return response.data;
+        return response.data.map((category : Category) => {
+            const gender : Gender | undefined = store.getState().auth.userSession?.gender;
```

Comment:
Maybe send gender as a parameter from the component (using `useAppSelector(selectGender)`)
Then you won't need to do `store.getState().auth.userSession?.gender`